crates/indexer/src/models/property_map.rs

```diff
@@ -0,0 +1,41 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Result;
+use serde_json::Value;
+use std::collections::HashMap;
+
+use crate::models::token_bcs_utils as utils;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct PropertyValue {
+    value: String,
+    typ: String,
+}
+
+pub fn create_property_value(typ: String, value: String) -> Result<PropertyValue> {
+    Ok(PropertyValue {
+        value: utils::convert_bcs_hex(typ.clone(), value.clone()).unwrap_or(value),
+        typ,
+    })
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct PropertyMap {
+    data: HashMap<String, PropertyValue>,
+}
+
+impl PropertyMap {
+    pub fn from_bsc_encode_str(val: Value) -> Option<PropertyMap> {
+        let mut pm = PropertyMap {
+            data: HashMap::new(),
+        };
+        let records: &Vec<Value> = val["map"]["data"].as_array()?;
```

Comment:
Records can be null.

Reply:
Or empty I mean. I don't think that we can assume map or data exists.

Reply:
Actually nvm this somehow works. 

I am getting back {"data": {}} though when there's no property. Can we directly return what's inside data?

Reply:
Also tested and it doesn't work as intended. It's always returning {"data": {}}, e.g. when testing with `{“map”: {“data”: [{“key”: “mint_number”, “value”: {“type”: “number”, “value”: “0x30783839”}}]}}`

Reply:
Quotation mark in your string is “, while the actually quotation mark should be ". If you replace the quotation mark with the correct one, it should work.